src/Sulu/Bundle/CategoryBundle/Entity/CategoryTranslation.php

```diff
@@ -111,10 +146,122 @@ public function setCategory(\Sulu\Bundle\CategoryBundle\Entity\Category $categor
     /**
      * Get category.
      *
-     * @return \Sulu\Bundle\CategoryBundle\Entity\Category
+     * @return Category
      */
     public function getCategory()
     {
         return $this->category;
     }
+
+    /**
+     * Add keyWord.
+     *
+     * @param KeyWord $keyWord
+     *
+     * @return Category
+     */
+    public function addKeyWord(KeyWord $keyWord)
+    {
+        $this->keyWords[] = $keyWord;
+
+        return $this;
+    }
+
+    /**
+     * Remove keyWord.
+     *
+     * @param KeyWord $keyWord
+     */
+    public function removeKeyWord(KeyWord $keyWord)
+    {
+        $this->keyWords->removeElement($keyWord);
+    }
+
+    /**
+     * Get keyWords.
+     *
+     * @return Collection
+     */
+    public function getKeyWords()
+    {
+        return $this->keyWords;
+    }
+
+    /**
+     * Returns true if given keyword already linked with the category.
+     *
+     * @return bool
+     */
+    public function hasKeyWord(KeyWord $keyWord)
+    {
+        return $this->getKeyWords()->exists(
+            function ($key, KeyWord $element) use ($keyWord) {
+                return $element->compareWith($keyWord);
```

Comment:
this functionality would then be moved to the repository because we shouldn't load all keywords if they are not needed.